src/main/java/org/opensrp/service/StockService.java

```diff
@@ -185,6 +218,108 @@ public void deleteStock(Long id) {
 	  return allStocks.findStocksByLocationId(servicePointId);
 	}
 
+	public CsvBulkImportDataSummary convertandPersistInventorydata(List<Map<String, String>> csvStocks, String userName) {
+		int rowCount = 0;
+		CsvBulkImportDataSummary csvBulkImportDataSummary = new CsvBulkImportDataSummary();
+		FailedRecordSummary failedRecordSummary;
+		List<FailedRecordSummary> failedRecordSummaries = new ArrayList<>();
+		Inventory inventory;
+		Integer totalRows = csvStocks.size();
+		Integer rowsProcessed = 0;
+
+		try {
+			failedRecordSummaries = validateInventoryData(csvStocks);
+		}
+		catch (ParseException e) {
```

Comment:
I will add the error in the `failedRecordSummaries` list in the catch block as done on Line no: 251 
In this way, it would ensure if there is an error or exception occurred during validation!
@githengi

Reply:
noted